pulp_ansible/app/serializers.py

```diff
@@ -808,11 +818,38 @@ def to_representation(self, value):
         return [{"name": x, "url": value[x]} for x in value]
 
 
-class AnsibleNamespaceMetadataSerializer(NoArtifactContentSerializer):
+class NamespaceRelatedFieldSerializer(RelatedFieldsBaseSerializer):
+    my_permissions = MyPermissionsField(source="*", read_only=True)
+
+
+class NamespaceValidationMixin:
+    def validate_name(self, name):
+        if not name:
+            raise ValidationError(detail={"name": _("Attribute 'name' is required")})
+        if not re.match(r"^[a-z0-9_]+$", name):
+            raise ValidationError(
+                detail={
+                    "name": _("Name can only contain lower case letters, underscores and numbers")
+                }
+            )
+        if len(name) <= 2:
+            raise ValidationError(detail={"name": _("Name must be longer than 2 characters")})
+        if name[0] in "0123456789_":
+            raise ValidationError(detail={"name": _("Name cannot begin with '_' or any number")})
+
+        return name
+
+
+class AnsibleNamespaceMetadataSerializer(NoArtifactContentSerializer, NamespaceValidationMixin):
     """
     A serializer for Namespaces.
     """
 
+    # Task on the serializer will always return none. It's just here to prevent the bindings
+    # from breaking.
+    task = serializers.SerializerMethodField()
```

Comment:
Why would the bindings break?
We generate them for each version individually anyway.
I'd rather not do such things.